doc/reference/misc.md

```diff
@@ -1777,6 +1777,21 @@ Please document me!
 
 Please document me!
 
+### string-subst
+::: tip usage
+```
+(string-subst str old new [count = #f])
+  str   := string
+  old   := string
+  new   := string
+  count := fixnum
+=> string
+```
+:::
+
+In the copy of str replace the string old with string new.
+Count controls the number of replacements (count < 0 equals to #f).
```

Comment:
Shouldn't negative count as "don't do it" (same as 0) as in Common Lisp rather than "do it an unlimited number of time" as in Python? i.e. the stop condition isn't `done = count` but `done >= count`. Makes more sense, if the algorithm is allowed to bail early and not return an error when the number of matches is less than `count`.

Reply:
In general, I find that Python has a piss poor API to copy in its details.

Reply:
I am in agreement here -- -1 should be the same as 0. We are not python, we have `#f` to indicate bottom.